composites/Plugin/SnippetEditor/components/SnippetEditor.js

```diff
@@ -0,0 +1,387 @@
+/* External dependencies */
+import React from "react";
+import styled from "styled-components";
+import { FormattedMessage } from "react-intl";
+import PropTypes from "prop-types";
+
+/* Internal dependencies */
+import SnippetPreview, { MODE_DESKTOP, MODE_MOBILE, MODES } from "../../SnippetPreview/components/SnippetPreview";
+import SnippetEditorFields from "./SnippetEditorFields";
+import { Button } from "../../Shared/components/Button";
+import SvgIcon from "../../Shared/components/SvgIcon";
+import colors from "../../../../style-guide/colors.json";
+import { lengthAssessmentShape, replacementVariablesShape } from "../constants";
+import FormattedScreenReaderMessage from "../../../../a11y/FormattedScreenReaderMessage";
+
+const SwitcherButton = Button.extend`
```

Comment:
This component should be documented and probably other styled components should be documented.